### PR TITLE
Minor Interface Changes

### PR DESCRIPTION
**Pull request by DavidKeaton**
So chaps and lasses, I have here another little PR ironing out small things that I can do at work. :-P
- Do not draw the vehicle facing indicator when looking around (via `;`), so now looking around isn't confusing and odd.
- Changed inverted color value of `i_black` and `i_dkgray` from `COLOR_BLACK` to `COLOR_WHITE` (with the former still `A_BOLD` mind you), so now examining pavement doesn't make it seem like the cursor vanished!

That's pretty much it!
-Davek

### CODE REVIEW

File: src/player.h [hunk 1]
```diff
@@ -1214,6 +1214,9 @@ class player : public Character, public JsonSerializer, public JsonDeserializer
             turn_died = (turn_died != -1) ? turn : turn_died;
         }
 
+        // are we looking around via ';'?
+        bool is_looking_around;
```
Comment (BevapDin) [inline src/player.h]: While I like a solution to that problem, this solution is really ugly (a public member just for one task). It should probably be in `game` as it does not really concern the player object. Or you could add it as a parameter to `game::draw_ter`, it is called directly from `game::look_around`, the only place where the non-default value would be needed.

**Comment by DavidKeaton** [comment]
Funnily enough I had it in game, but moved it. I concur wholeheartedly on it needing to be moved. Anything else that I should touch up so I can sweep it up in one pass?

**Comment by BevapDin** [comment]
> Also considered new parameters for function, but unsure how you guys feel about that.

Personally, I recommend overloading the functions like this:

```C++
game {
  draw_ter(int, int); // old, but all code will still use it.
  draw_ter(int, int, bool); // new, but keep the above one
};

// add the new parameter here:
game::draw_ter(int, int) { ... }
// -> becomes:
game::draw_ter(int, int, bool) { ... }

// add an overloaded function that forwards to the new one:
game::draw_ter(int, int) {
  return draw_ter(x,y, true); // add the default value here.
}
```
This hides the implementation and the value of the default parameter (or even its calculation). You can even make the new version with the boolean parameter private or protected.

Otherwise: adding new parameters should not be a problem, but what do I know.

**Comment by DavidKeaton** [comment]
Oh man would you prefer I overload and wrap? I simply just put a defaulted boolean argument instead. :-S

**Comment by DavidKeaton** [comment]
Also you know a lot matey. :-)

**Comment by BevapDin** [comment]
In my experience: defaulted values sometimes cause trouble as the are not obvious. Overloading "usually" don't cause that much trouble.
Defaulted parameters expose the default value to the caller, so they can not be changed later as the caller might have incorporated the known default value in its own logic.

```C++
struct Something {
    void doit(int what = 1);
};

void abc() {
    what = json.get_int( "what", 1 ); // 1 is default to do_something, so we use it here as default as well.
    something.doit(what);
}
```
Later the default of `doit` is changed and the whole thing breaks.

Or add a `default_what` to `Something`:
```C++
struct Something {
    void doit(int what = default_what); // won't compile
private:
    int default_what;
};
```
Or add a complex calculation of the default value maybe even depending on the other (non-default) parameters...

----

But this all very subjective. Both techniques are valid and useful. *shrug*

**Comment by DavidKeaton** [comment]
Well seeing as you are clearly above my programmatic expertise, I will opt to learn from you as opposed to blazing my own way and wrecking catastrophically. I will consider you my one stop shop for C++ ownage.

Regardless, it is now wrapped appropriately and Syntastic seems to throw no errors. Pushing my commit for your review good sir.

**Comment by DavidKeaton** [comment]
Hello again Mr. @KA101 :-)

**Comment by KA101** [comment]
There's no need to ping me, I see every time someone posts.

(Would that such included new commits, too.)

**Comment by DavidKeaton** [comment]
Ah I didn't even think about the pinging. My bad.

**Comment by KA101** [comment]
OK, I'm willing to merge this

BUT

I don't use the veh direction indicator.  It's possible that folks driving small vehicles and looking around might want the indicator (or some other indicator) to tell 'em which way the veh's going if they look around for a while and lose track of that detail.

Just something to keep in mind for future, not a revert demand now.

**Pull request merged**